### PR TITLE
Removes non-combat 'click delay'

### PR DESCRIPTION
**Pull request by carlarctg**
# About the pull request

Removed next move addition from these things:
- Picking things up from the ground
- Yanking out objects (???)
- Transferring from stack to stack (????)
- Placing things on tables
- Item attack delay has been removed from click_adjacent() and moved to attackby() (means you'll only get that click delay when attacking something)

Clicking on the floor (open turfs) with an item will cause next_move addition to prevent this PR from impacting combat. 
Additionally, it will display a to_chat that can be disabled when disabling fluff actions in chat logs. This will also happen with fists on disarm or harm intent.

Whacking random structures will no longer display a hitting animation.

# Explain why it's good for the game

Every single other SS13 server has removed this, including, Stan hear me out, TG. The only thing this does is be an extremely fucking annoying nuisance that hinders anything that isn't shooting things, worsening the life of researchers, MTs, MSTs, Req (of course), and all other staff to various degrees. Even if it's 'realistic' that you can't quickly pick up and put down things (which isn't true, you can do so in a rush IRL), it's just a thoroughly unpleasant experience to sit there staring at the screen because you slightly misclicked/maybe even if you didn't, it makes everything slow and annoying.

This isn't an evil QOL addition that works towards the slow erosion of anything mechanically complex in the game, it's just updating the fucking game to 2023 SS13 server standards.

And yes, I've tested it. This won't result in infinite extremely fast knife stabs that kill xenos in a second.

**THIS ABSOLUTELY MUST BE TESTMERGED BEFORE MERGING**

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
del: Removed non-combat 'click delay', interacting with items not on your person should be significantly smoother.
balance: Clicking on the floor (open turfs) with an item will cause next_move addition to prevent this PR from impacting combat. 
Additionally, it will display a to_chat that can be disabled when disabling fluff actions in chat logs. This will also happen with fists on disarm or harm intent. (This is pre-existing behaviour, all that is changed here is the addition of the message)
del: Whacking random structures will no longer display a useless hitting animation.
/:cl:

### CODE REVIEW

**Comment by stanalbatross** [comment]
`And yes, I've tested it. This won't result in infinite extremely fast knife stabs that kill xenos in a second.`

well clearly you didn't test on closed turfs (i.e. resin walls) because there's no delay on attacking those

**Comment by carlarctg** [comment]
> `And yes, I've tested it. This won't result in infinite extremely fast knife stabs that kill xenos in a second.`
> 
> well clearly you didn't test on closed turfs (i.e. resin walls) because there's no delay on attacking those

that's what testmerges are for

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -465,3 +465,11 @@
 	for(var/h in src.hud_possible)
 		src.clone.hud_list[h].icon_state = src.hud_list[h].icon_state
 
+/mob/living/attack_hand(mob/user) // not recommended to be used - does not apply next move cooldown by default (how do i dmdoc procs?)
```
Comment (harryob) [inline code/modules/mob/living/living.dm:468]: https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md#documenting-a-proc

**Comment by burgz1** [comment]
i'm no programmer, and i've never used github before. But i want to mention that the stunbaton is broken with this merge. you can just instantly stun people.

**Comment by carlarctg** [comment]
ready for TM

**Comment by Herotitan772** [comment]
as a frequent player of Engineer, medic, doctor, researcher and OT this is the best thing since sliced bread. +1

**Comment by Superwelder** [comment]
I am not 100% certain its related to this issue but now tables with a corner can be flipped and its causing a lot of headaches. For one, it makes no sense that a corner table can be magically separated into its horizontals and verticals when the sprite clearly shows it as connected. 2nd, it makes accidental table flips SUPER common because you now can accidentally flip any table. Please fix this! It also has slight balance implications because xenos can abuse the corner tables to flip and move into an area that clearly looks like movement would be blocked until the table was vaulted or smashed(flipping is instant).

**Comment by harryob** [comment]
> I am not 100% certain its related to this merge but now tables with a corner can be flipped in any direction and its causing a lot of headaches. For one, it makes no sense that a corner table can be magically separated into its horizontals and verticals when the sprite clearly shows it as connected. 2nd, it makes accidental table flips SUPER common because you now can accidentally flip any table. Please fix this! It also has slight balance implications because xenos can abuse the corner tables to flip and move into an area that clearly looks like movement would be blocked until the table was vaulted or smashed(flipping is instant).

@Superwelder 
this is a bug in master, not this pr. feel free to make the issue report

**Comment by Herotitan772** [comment]
This doesn’t work for engineering, mainly wiring cades, adding metal/plas to girders, unscrewing stuff, deconing stuff, building stuff, hacking stuff or welding stuff

**Comment by Herotitan772** [comment]
can this get merged into master?

File: code/game/objects/structures.dm [hunk 2]
```diff
@@ -38,12 +38,18 @@
...
 
 /obj/structure/attackby(obj/item/W, mob/user)
+	if(!(W.flags_item & ITEM_ABSTRACT))
+		user.next_move += W.attack_speed
 	if(HAS_TRAIT(W, TRAIT_TOOL_WRENCH))
 		if(user.action_busy)
 			return TRUE
-		toggle_anchored(W, user)
-		return TRUE
-	..()
```
Comment (Watermelon914) [inline code/game/objects/structures.dm:46]: Why remove the parent call? I think this should still be calling parent

File: code/game/objects/structures/surface.dm [hunk 3]
```diff
@@ -137,12 +137,15 @@
 		detach_item(O)
 
 /obj/structure/surface/attackby(obj/item/W, mob/user, click_data)
+	after_attack_by(W, user, click_data)
+	return
```
Comment (Watermelon914) [inline code/game/objects/structures/surface.dm:141]: You can remove this return.

File: code/game/turfs/floor.dm [hunk 4]
```diff
@@ -169,8 +169,19 @@
...
 
+	if(!(hitting_item.flags_item & NOBLUDGEON))
+		user.next_move += hitting_item.attack_speed
+	if(!(hitting_item.flags_item & ITEM_ABSTRACT))
+		user.show_message(SPAN_NOTICE("You swing \the [hitting_item] through the air."), message_flags = CHAT_TYPE_FLUFF_ACTION)
+
 	return ..()
 
+/turf/open/floor/attack_hand(mob/user)
+	if(!user.a_intent == INTENT_HARM|INTENT_DISARM)
```
Comment (Watermelon914) [inline code/game/turfs/floor.dm:180]:
```suggestion
	if(!user.a_intent == (INTENT_HARM|INTENT_DISARM))
```

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
Generally speaking, I think this implementation is flawed and you should really be having "after_attack_by" at the root level, aka /atom
Then, you should move all attackby calls to an override of after_attack_by whilst keeping attackby set to SHOULD_CALL_PARENT(TRUE) so that after_attack_by always gets called during attackby.
The current implementation you've done requires multiple definitions for after_attack_by on different types, which is non-ideal for maintainability.

**Pull request closed without merging**